src/Repository/DefaultRepository.php

```diff
@@ -129,7 +129,7 @@ static function (object $event) use ($aggregate, &$playhead, $messageDecorator)
                     ->withAggregateClass($aggregate::class)
                     ->withAggregateId($aggregate->aggregateRootId())
                     ->withPlayhead(++$playhead)
-                    ->withRecordedOn(Clock::createDateTimeImmutable());
+                    ->withRecordedOn(new DateTimeImmutable());
```

Comment:
i dont know how we should handle this

Reply:
i think you can remove this line

Reply:
And you need add the RecordedOnDecorator as default.

Reply:
Yeah but where? Should we add this at the manager if no decorator is given? What happen then if a decorator is given but no `RecordedOnDecorator` is in the chain e.g.

Reply:
I would set it as default in the repository. If `RecordedOnDecorator ` is not set, then there is an error when writing that the RecordedOn header could not be found.

All systems with default middleware or decorators have the problem that the user may overwrite them or forget to include them. eg. with the message component you can also forget about the HandleMiddleware. And then the whole component no longer makes sense.